File: example/javalib/module/4-programmable-config/custom-src/Foo2.java
package foo;

import java.io.IOException;
import java.util.Properties;
import org.thymeleaf.TemplateEngine;
import org.thymeleaf.context.Context;

public class Foo2 {
  public static final String value() {
    Context context = new Context();
    context.setVariable("text", "hello2");
    return new TemplateEngine().process("<h1 th:text=\"${text}\"></h1>", context);
  }

  public static void main(String[] args) {
    System.out.println("Foo2.value: " + Foo2.value());
    System.out.println("Foo.value: " + Foo.value());

    System.out.println("FooA.value: " + FooA.value);
    System.out.println("FooB.value: " + FooB.value);
    System.out.println("FooC.value: " + FooC.value);

    System.out.println("MyResource: " + readResource("MyResource.txt"));
    System.out.println("MyOtherResource: " + readResource("MyOtherResource.txt"));

    Properties properties = System.getProperties();
    System.out.println("my.custom.property: " + properties.getProperty("my.custom.property"));

    String myCustomEnv = System.getenv("MY_CUSTOM_ENV");
    if (myCustomEnv != null) {
      System.out.println("MY_CUSTOM_ENV: " + myCustomEnv);
    }
  }

  private static String readResource(String resourceName) {
    try {
      return new String(
          Foo2.class.getClassLoader().getResourceAsStream(resourceName).readAllBytes());
    } catch (IOException e) {
      return null;
    }
  }
}

File: example/kotlinlib/module/1-common-config/custom-src/Foo2.kt
package foo

import java.io.BufferedReader
import java.io.IOException
import java.io.InputStream
import java.io.InputStreamReader
import java.util.Properties
import kotlinx.html.h1
import kotlinx.html.stream.createHTML

fun main(args: Array<String>) {
    println("Foo2.value: ${Foo2.VALUE}")
    println("Foo.value: ${Foo.VALUE}")

    println("FooA.value: ${FooA.VALUE}")
    println("FooB.value: ${FooB.VALUE}")
    println("FooC.value: ${FooC.VALUE}")

    println("MyResource: ${readResource("MyResource.txt")}")
    println("MyOtherResource: ${readResource("MyOtherResource.txt")}")

    val properties = System.getProperties()
    println("my.custom.property: ${properties.getProperty("my.custom.property")}")

    val myCustomEnv = System.getenv("MY_CUSTOM_ENV")
    if (myCustomEnv != null) {
        println("MY_CUSTOM_ENV: $myCustomEnv")
    }
}

object Foo2 {
    val VALUE = createHTML(prettyPrint = false).h1 { text("hello2")  }.toString()
}

private fun readResource(resourceName: String): String? {
    return try {
        ::main.javaClass.classLoader
            .getResourceAsStream(resourceName)
            .readAllBytes()
            .toString(Charsets.UTF_8)
    } catch (e: IOException) {
        null
    }
}
